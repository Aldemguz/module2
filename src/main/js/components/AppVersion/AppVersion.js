import React from "react"
import {connect} from "react-redux";
import {withTranslation} from "react-i18next";
import {bindActionCreators, compose} from "redux";
import {getAppInfo} from "bucares/actions/appActions";

export class AppVersion extends React.Component {

    componentDidMount(){
        this.retrieveDataFromServer();
    }

    retrieveDataFromServer() {
        const {app} = this.props;
        if (app.appInfo === null && !app.appInfoLoading) {
            this.props.getAppInfo();
        }
    }

    render() {
        const {t, app} = this.props;
        return (
            <div>
                {t("version")}: {app.appInfo !== null ? app.appInfo.version : t("none")}
            </div>
        )
    }
}

function mapDispatchToProps(dispatch) {
    return bindActionCreators({
        getAppInfo: getAppInfo
    }, dispatch)
}

export default compose(
    connect((store) => ({app: store.get("app")}), mapDispatchToProps),
    withTranslation("translation")
)(AppVersion);
