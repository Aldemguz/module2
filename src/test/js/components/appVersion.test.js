import React from 'react';
import AppVersion from 'bucares/components/AppVersion';

describe('AppVersion Component Test', () => {

    it('Snapshot AppVersion', () => {
        const store = mockStore({
            get : (property) => {
                const reducers = {
                    app : {
                        appInfo: {
                            version : "1.0.0"
                        },
                        appInfoLoading: false
                    }
                };
                return reducers[property];
            }
        });

        const appVersion = global.mount(
            globalWrapper(<AppVersion/>, store)
        );
        expect(getJSON(appVersion)).toMatchSnapshot()
    });

    it('Snapshot AppVersion with appInfo null', () => {
        const store = mockStore({
            get : (property) => {
                const reducers = {
                    app : {
                        appInfo: null,
                        appInfoLoading: false
                    }
                };
                return reducers[property];
            }
        });

        const appVersion = global.mount(
            globalWrapper(<AppVersion/>, store)
        );
        expect(getJSON(appVersion)).toMatchSnapshot()
    });
});