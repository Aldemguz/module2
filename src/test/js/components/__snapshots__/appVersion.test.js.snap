// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppVersion Component Test Snapshot AppVersion 1`] = `
<Connect(withI18nextTranslation(AppVersion))>
  <div>
    version
    : 
    1.0.0
  </div>
</Connect(withI18nextTranslation(AppVersion))>
`;

exports[`AppVersion Component Test Snapshot AppVersion with appInfo null 1`] = `
<Connect(withI18nextTranslation(AppVersion))>
  <div>
    version
    : 
    none
  </div>
</Connect(withI18nextTranslation(AppVersion))>
`;
